pkg/resolvers/endpoints.go

```diff
@@ -321,25 +335,25 @@ func (r *defaultEndpointsResolver) getMatchingPodAddresses(ctx context.Context,
 		return nil
 	}
 	r.logger.V(1).Info("Got pods for label selector", "count", len(podList.Items), "selector", ls.String())
+
 	for _, pod := range podList.Items {
 		podIP := k8s.GetPodIP(&pod)
 		if len(podIP) == 0 {
 			r.logger.Info("pod IP not assigned yet", "pod", k8s.NamespacedName(&pod))
 			continue
 		}
-		portList := r.getPortList(pod, ports)
-		if len(ports) != len(portList) && len(portList) == 0 {
-			r.logger.Info("Couldn't get matched port list from the pod", "pod", k8s.NamespacedName(&pod), "expectedPorts", ports)
-			continue
+
+		if policyType == networking.PolicyTypeEgress {
+			portList = r.getPortList(pod, rulePorts)
```

Comment:
Why are we overwriting the `portList` here? Shouldn't we append instead? we can have the same pod selectors match on multiple unique pods/deployments. I see we're appending for the ingress but not here..

Reply:
For ingress we aggregate ports from ingress rule together and apply the same set to individual pod ip. But for egress, we are setting individual entry for every pod/ip with their allowed port(s). I feel we shouldn't combine ports for egress entry.

Reply:
Makes sense, as we're adding port list per pod IP ..